Scripts/Runtime/Jobs/Collections/CollectionAccessControlUtil.cs

```diff
@@ -0,0 +1,191 @@
+        /// enums for example.
+        /// This is a child of <see cref="WorldLookup"/>
+        /// </summary>
+        private class TypeLookup : AbstractLookup<World, Type, IContextLookup>
+        {
+            private static IContextLookup CreationFunction<TContext>(Type context)
+            {
+                return new ContextLookup<TContext>(context);
+            }
+
+            internal TypeLookup(World context) : base(context)
+            {
+            }
+
+            internal CollectionAccessController<TContext> GetOrCreate<TContext>(TContext context)
+            {
+                Type contextType = typeof(TContext);
+                ContextLookup<TContext> contextLookup = (ContextLookup<TContext>)LookupGetOrCreate(contextType, CreationFunction<TContext>);
+                return contextLookup.GetOrCreate(context);
+            }
+
+            internal void Remove<TContext>(TContext context)
+            {
+                Type contextType = typeof(TContext);
+                if (!TryGet(contextType, out IContextLookup contextLookup))
+                {
+                    return;
+                }
+
+                ((ContextLookup<TContext>)contextLookup).Remove(context);
+            }
+        }
+
+        /// <summary>
+        /// Lookup based on a specific value of a <see cref="Type"/> from the parent <see cref="TypeLookup"/>
+        /// Allows for a <see cref="CollectionAccessController{TContext}"/> to be specific to a value of an enum for
+        /// example.
+        /// </summary>
+        internal class ContextLookup<TContext> : AbstractLookup<Type, TContext, ICollectionAccessController>,
+                                                 IContextLookup
+        {
+            internal ContextLookup(Type context) : base(context)
+            {
+            }
+
+            internal CollectionAccessController<TContext> GetOrCreate(TContext context)
+            {
+                return (CollectionAccessController<TContext>)LookupGetOrCreate(context, CreationFunction);
+            }
+
+            private ICollectionAccessController CreationFunction(TContext context)
+            {
+                return new CollectionAccessController<TContext>(context, this);
+            }
+
+            internal void Remove(TContext context)
+            {
+                if (!ContainsKey(context))
+                {
+                    return;
+                }
+
+                LookupRemove(context);
+            }
+        }
+
+
+        //*************************************************************************************************************
+        // PUBLIC STATIC API
+        //*************************************************************************************************************
+
+        private static WorldLookup s_WorldLookup;
+
+        private static WorldLookup Lookup
+        {
+            get => s_WorldLookup ?? (s_WorldLookup = new WorldLookup());
+        }
```

Comment:
This vs `WorldCacheUtil` is confusing. Can they be merged? If not are there more descriptive names that could be used?

Reply:
Changed to `LookupByWorld`. 

Also changed the names in the other Managers. 

So now you have:

`TopLevelManager`
 - `LookupByWorld`
     - `LookupByType`
         - `LookupByX`

Reply:
Not as relevant anymore after refactor to systems